### PR TITLE
Chapter 2 tasks

### PR DESCRIPTION
**Pull request by sdimitrijevikj**

### CODE REVIEW

**Comment by sdimitrijevikj** [comment]
cc @gillchristian :)

File: src/Chapter2.hs [hunk 1]
```diff
@@ -336,7 +348,9 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList start end list
+    | start < 0 || end < 0 = []
+    | otherwise = take (end - start + 1) (drop start list)
```
Comment (gillchristian) [inline src/Chapter2.hs:353]: 😏

File: src/Chapter2.hs [hunk 2]
```diff
@@ -606,7 +624,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate [] = []
+duplicate (x:xs) = [x, x] ++ duplicate(xs)
```
Comment (gillchristian) [inline src/Chapter2.hs:628]: No need for parenthesis in this case

```suggestion
duplicate (x:xs) = [x, x] ++ duplicate xs
```

File: src/Chapter2.hs [hunk 3]
```diff
@@ -621,7 +640,17 @@ Write a function that takes elements of a list only in even positions.
...
+takeEven [] = []
+takeEven [_] = [1]
+takeEven [_, _] = [1]
+takeEven l = go [] l
+    where
+        go :: [Int] -> [Int] -> [Int]
+        go acc [] = acc
+        go acc (x:xs)
+            | mod (length xs) 2 == 0 = go (acc ++ [x]) xs
+            | otherwise = go acc xs
```
Comment (gillchristian) [inline src/Chapter2.hs:653]: This function has to keep the elements that are on odd positions, so if I understand correctly it should work on any type of element:

```hs
takeEven [2, 1, 3, 5, 4]
--        ^  ^  ^  ^  ^
--        0  1  2  3  4
-- keeps the position 0, 2, & 4
[2, 3, 4]

takeEven [True, False, True, False, True]
--        ^     ^      ^     ^      ^
--        0     1      2     3      4
[True, True, True]

takeEven ["a", "b", "c", "d", "e"]
--        ^    ^    ^    ^    ^
--        0    1    2    3    4
["a", "c", "e"]
```

File: src/Chapter2.hs [hunk 4]
```diff
@@ -728,7 +757,7 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concat (map (\x -> replicate x x) l)
```
Comment (gillchristian) [inline src/Chapter2.hs:760]: 🤯

File: src/Chapter2.hs [hunk 5]
```diff
@@ -781,13 +811,13 @@ Let's now try to eta-reduce some of the functions and ensure that we
...
-divideTenBy x = div 10 x
+divideTenBy = div 10
 
--- TODO: type ;)
+listElementsLessThan :: Int -> [Int] -> [Int]
 listElementsLessThan x l = filter (< x) l
 
 -- Can you eta-reduce this one???
-pairMul xs ys = zipWith (*) xs ys
+pairMul = zipWith (*)
```
Comment (gillchristian) [inline src/Chapter2.hs:820]: Let's silence the linter here by adding the signature 🤫

**Review by gillchristian — COMMENTED** [review]
Nice work! 👏

**Review by gillchristian — APPROVED** [review]
:rocket:

**Pull request merged**